Lib/test/test_peepholer.py

```diff
@@ -494,6 +495,29 @@ def genexpr():
             return (y for x in a for y in [f(x)])
         self.assertEqual(count_instr_recursively(genexpr, 'FOR_ITER'), 1)
 
+    def test_format(self):
+        flags = '-+ #0'
+        testcases = [
+            *product(('', '1234',), 'sra'),
```

Comment:
Test case(s) that include other % formatting directives within the value would be good.

We want to make sure injection attacks don't exist if this happened to get implemented in the wrong serialized actions manner.

```
untrusted = '%% %s'
value = 'foo %s bar' % untrusted
```

results in `value == 'foo %% %s bar'` rather than raising an exception about tuple size or converting %% into %.

This might make a good test of its own rather than shoehorning it into this one given some complexities of where % directives that'll be peepholed can appear in strings and in combination with others that may or may not be.

Reply:
Added more tests. Caught and fixed conversion of non-numbers to int and float (`'%d' % '123'` did not raise exception before).